src/query/api/v1/handler/prometheus/remote/write_test.go

```diff
@@ -397,6 +382,38 @@ func TestPromWriteGraphiteMetricsTypes(t *testing.T) {
 	require.NoError(t, capturedIter.Error())
 }
 
+func TestPromWriteDisabledMetricsTypes(t *testing.T) {
+	ctrl := xtest.NewController(t)
+	defer ctrl.Finish()
+
+	var capturedIter ingest.DownsampleAndWriteIter
+	mockDownsamplerAndWriter := ingest.NewMockDownsamplerAndWriter(ctrl)
+	mockDownsamplerAndWriter.
+		EXPECT().
+		WriteBatch(gomock.Any(), gomock.Any(), gomock.Any()).
+		Do(func(_ context.Context, iter ingest.DownsampleAndWriteIter, _ ingest.WriteOptions) ingest.BatchError {
+			capturedIter = iter
+			return nil
+		})
+
+	opts := makeOptions(mockDownsamplerAndWriter).SetStoreMetricsType(false)
+
+	promReq := &prompb.WriteRequest{
+		Timeseries: []prompb.TimeSeries{
+			{Type: prompb.MetricType_COUNTER},
+		},
+	}
```

Comment:
Mind adding a test with no type / invalid type added to ensure it will be correctly ignored with `SetMetricsType == false`?

Reply:
I'll add a case with no type.
Invalid protobuf values trigger an error here (as well as in prior conversions from protobuf on these endpoints).